crates/node/src/sync/key.rs

```diff
@@ -33,6 +33,30 @@ impl Node {
             bail!("connection closed while awaiting state sync handshake");
         };
 
+        let their_identity = match ack {
+            StreamMessage::Init {
+                party_id,
+                payload: InitPayload::KeyShare,
+                ..
+            } => party_id,
+            unexpected @ (StreamMessage::Init { .. }
+            | StreamMessage::Message { .. }
+            | StreamMessage::OpaqueError) => {
+                bail!("unexpected message: {:?}", unexpected)
+            }
+        };
+
+        let private_key = self
+            .ctx_manager
+            .get_private_key(context.id, our_identity)?
+            .ok_or_eyre("expected own identity to have private key")?;
+
+        let shared_key = SharedKey::new(&private_key, &their_identity);
+
+        let Some(ack) = recv(stream, self.sync_config.timeout, Some(shared_key)).await? else {
+            bail!("connection closed while awaiting state sync handshake");
+        };
+
         let sender_key = match ack {
```

Comment:
we're done with the handshake, now we're sending the messages

```suggestion
        let Some(msg) = recv(stream, self.sync_config.timeout, Some(shared_key)).await? else {
            bail!("connection closed while awaiting key share");
        };

        let sender_key = match msg {
```